extensions/amp-story/1.0/media-pool.js

```diff
@@ -514,25 +514,31 @@ export class MediaPool {
     return this.enqueueMediaElementTask_(
       poolMediaEl,
       new SwapIntoDomTask(placeholderEl)
-    ).then(
-      () => {
-        this.enqueueMediaElementTask_(
-          poolMediaEl,
-          new UpdateSourcesTask(this.win_, sources)
-        );
-        this.enqueueMediaElementTask_(poolMediaEl, new LoadTask()).then(() => {
-          this.maybeResetAmpMedia_(ampMediaForPoolEl);
-          this.maybeResetAmpMedia_(ampMediaForDomEl);
-        });
-      },
-      () => {
-        this.forceDeallocateMediaElement_(poolMediaEl);
-      }
-    );
+    )
+      .then(() =>
+        Promise.all([
+          this.maybeResetAmpMedia_(ampMediaForPoolEl),
+          this.maybeResetAmpMedia_(ampMediaForDomEl),
+        ])
+      )
+      .then(
+        () => {
+          const sourcesPromise = this.enqueueMediaElementTask_(
```

Comment:
Return sources so `playAllMedia` can play videos when they have the sources. Otherwise `loadPromise` would fail.

Reply:
let's write this code in the most predictable way and wait for all actions to be done before its promise resolves:

```
.then(() => enqueue UpdateSourcesTask)
.then(() => enqueue LoadTask)
```